eos-grub-upgrade

```diff
@@ -0,0 +1,276 @@
+# +------+------------------------------------------+---------------+---------------------------------------------------------------+
+
+ESP_MOUNTPOINT=/boot/efi
+
+##########################################################################
+## UEFI
+##########################################################################
+
+## AMD64 GRUB
+declare -A GRUB_AMD64=(
+  [A]="${ESP_MOUNTPOINT}/EFI/endless/grubx64.efi"
+  [B]="${ESP_MOUNTPOINT}/EFI/BOOT/grubx64.efi"
+  [C]="${ESP_MOUNTPOINT}/EFI/BOOT/bootx64.efi"
+)
+NEW_GRUB_AMD64=/usr/lib/grub/images/x86_64-efi/grubx64.efi
+
+## SHIM
+declare -A SHIM=(
+  [A]="${ESP_MOUNTPOINT}/EFI/BOOT/bootx64.efi"
+  [B]="${ESP_MOUNTPOINT}/EFI/BOOT/bootx64.efi"
+)
+NEW_SHIM=/usr/lib/shim/shim.efi
+
+SHIM_COPY=${ESP_MOUNTPOINT}/EFI/endless/shim.efi
+NEW_SHIM_COPY=/usr/lib/shim/shim.efi
+
+## IA32 GRUB
+GRUB_IA32=${ESP_MOUNTPOINT}/EFI/BOOT/bootia32.efi
+NEW_GRUB_IA32=/usr/lib/grub/images/i386-efi/bootia32.efi
+
+## MokManager
+MOKMANAGER=${ESP_MOUNTPOINT}/EFI/endless/MokManager.efi
+NEW_MOKMANAGER=/usr/lib/shim/MokManager.efi
+
+## fallback
+FALLBACK=${ESP_MOUNTPOINT}/EFI/BOOT/fallback.efi
+NEW_FALLBACK=/usr/lib/shim/fallback.efi
+
+## BOOT.CSV
+BOOTCSV=${ESP_MOUNTPOINT}/EFI/endless/BOOT.CSV
+NEW_BOOTCSV=/usr/lib/shim/BOOT.CSV
+
+## grub.cfg
+GRUB_CFG=/boot/grub/grub.cfg
+NEW_GRUB_CFG=/usr/lib/grub/conf/grub.cfg
+
+EFI_LIST=( GRUB_AMD64 SHIM SHIM_COPY GRUB_IA32 MOKMANAGER FALLBACK BOOTCSV GRUB_CFG )
+
+##########################################################################
+
+umount_esp() {
+  umount ${ESP_MOUNTPOINT} 2> /dev/null
+}
+trap umount_esp EXIT
+
+get_last_char() {
+  local ret=$1
+
+  while [[ ${#ret} != "1" ]]; do
+    ret=${ret#?}
+  done
+
+  echo "${ret}"
+}
+
+get_root_disk() {
+  local root_part
+  root_part=$(systemctl show -p What sysroot.mount)
+  root_part=${root_part#What=}
+
+  if [[ -z ${root_part} ]]; then
+    echo "Couldn't identify root device"
+    exit 1
+  fi
+
+  # Get the root disk device file
+  local root_disk=
+  case ${root_part} in
+    /dev/sd??)
+      root_disk=${root_part%?}
+      ;;
+    /dev/*p[0-9])
+      root_disk=${root_part%p?}
+      ;;
+  esac
+
+  if [[ -z ${root_disk} ]]; then
+    echo "No root disk found for ${root_part}"
+    exit 1
+  fi
+
+  echo "${root_disk}"
+}
+
+# Returns "gpt" or "dos"
+get_partition_table() {
+  local root_disk=$1
+
+  local pt_label=$(blkid -o value -s PTTYPE "${root_disk}")
+  if [[ -z ${pt_label} ]]; then
+    echo "blkid -o value -s PTTYPE '${root_disk}' failed"
+    exit 1
+  fi
+
+  echo "${pt_label}"
+}
+
+get_firmware_type() {
+  if [[ -d /sys/firmware/efi ]]; then
+    echo "uefi"
+  else
+    echo "bios"
+  fi
+}
+
+mount_ESP() {
+  local root_disk=$1
+  local esp_guid="C12A7328-F81F-11D2-BA4B-00A0C93EC93B"
+
+  # Check if ESP does exist and retrieve the partition name
+  local esp_part=$(sfdisk -d "${root_disk}" | grep -i type=${esp_guid} | sed 's/ .*//')
```

Comment:
`sfdisk --dump "$root_disk" | awk '/type='$esp_guid'/ { print $1 }'` is a little neater than `| grep | sed` if we really must parse the output of `sfdisk` with shell pipelines.